From 26b1363c7cbf5d01c730f45fc0b4abc5686beaf7 Mon Sep 17 00:00:00 2001
From: Damien Miller <djm@mindrot.org>
Date: Sun, 18 Oct 2026 19:23:55 +0800
Subject: [PATCH] allow restart_syscall(2) in seccomp sandbox

bz4036, from Chester Weber
ok dtucker@

(cherry picked from commit 26b1363c7cbf5d01c730f45fc0b4abc5686beaf7)
---
 sandbox-seccomp-filter.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sandbox-seccomp-filter.c b/sandbox-seccomp-filter.c
index f696d50..b3bce68 100644
--- a/sandbox-seccomp-filter.c
+++ b/sandbox-seccomp-filter.c
@@ -441,6 +441,9 @@ static const struct sock_filter preauth_insns[] = {
 #ifdef __NR_uname
 	SC_ALLOW(__NR_uname),
 #endif
+#ifdef __NR_restart_syscall
+	SC_ALLOW(__NR_restart_syscall),
+#endif
 #ifdef __NR_setsockopt
 	SC_ALLOW_SETSOCKOPT(IPPROTO_IPV6, IPV6_TCLASS),
 	SC_ALLOW_SETSOCKOPT(IPPROTO_IP, IP_TOS),
-- 
2.53.0

